run_tests.py

```diff
@@ -0,0 +1,284 @@
+
+    return grouped_results
+
+
+def has_failed_tests(results):
+    for result in results:
+        if result['status'] == "failed":
+            return True
+
+    return False
+
+
+# Read 'com.apple.progress.fractionCompleted' extended file attribute from
+# ouput file to display the test runner progress.
+def watch_test_runner_progress(file_path, timeout=30):
+    progress = 0
+    last_progress_time = time.time()
+
+    print("Waiting for file to be created…")
+    # Wait for output file to be available on disk. This file is written
+    # at the beginning of the integration tests plugin run.
+    while not os.path.exists(file_path):
+        time.sleep(2)
+
+        if (time.time() - last_progress_time > timeout):
+            raise Exception("Timeout")
+
+    if not os.path.isfile(file_path):
+        raise Exception("Path to test output is not a file")
+
+    print("Waiting for tests to complete…")
+    while progress < 1:
+        stream = subprocess.Popen(
+            ['xattr', '-l', file_path],
+            stdout=subprocess.PIPE)
+
+        output = str(stream.communicate())
+        output = output.split("com.apple.progress.fractionCompleted:")[1]
+        match = re.search('\d+(.\d+)*', output)
+
+        latest_progress = float(match.group(0))
+        if latest_progress == progress:
+            # no progress since last run
+            if (time.time() - last_progress_time > timeout):
+                raise Exception("Timeout")
+
+            time.sleep(1)
+            continue
+
+        progress = latest_progress
+        print(f"Running tests: {round(progress * 100, 2)}% complete", end='\r')
+
+        time.sleep(1)
+        last_progress_time = time.time()
+
+
+def print_results(results):
+    
+    for suite_name, suite_results in results.items():
+
+        suite_status = "failed" if has_failed_tests(suite_results['results']) else "passed"
+
+        print('\n:: {status}\t{name} {relativePath}'.format(
+            status=suite_status.upper(),
+            name=suite_name,
+            relativePath=suite_results['relativePath']
+        ))
+
+        for res in suite_results['results']:
+            ancestors = " › ".join(res.get('ancestorTitles', []))
+            print(' • {status}\t{ancestors} {title}'.format(
+                status=res['status'].upper(),
+                ancestors=ancestors,
+                title=res['title'],
+            ))
+
+            if 'failureReason' not in res:
+                continue
+
+            failure_reason = '{}\n'.format(re.sub(
+                '{{{((\w*)|(\/\w*))}}}', '', res['failureReason']['message']))
+            print('\t {failure_reason}'.format(failure_reason=failure_reason))
+
+
+def parse_test_results(file_path):
+    if not os.path.isfile(file_path):
+        raise Exception('File not found')
+
+    # Open the output file and parse the results
+    with open(file_path, 'r') as f:
+        data = f.read()
```

Comment:
These two lines can be combined into one - just use json.load() 

Then you can move the rest of the code in the with() up one level